spec/chore/queues/sqs/consumer_spec.rb

```diff
@@ -1,145 +1,194 @@
-require 'spec_helper'
-
-describe Chore::Queues::SQS::Consumer do
-  let(:queue_name) { "test" }
-  let(:queue_url) { "test_url" }
-  let(:queues) { double("queues") }
-  let(:queue) { double("test_queue", :visibility_timeout=>10, :url=>"test_queue", :name=>"test_queue") }
-  let(:options) { {} }
-  let(:consumer) { Chore::Queues::SQS::Consumer.new(queue_name) }
-  let(:message) { TestMessage.new("handle",queue, "message body", 1) }
-  let(:message_data) {{:id=>message.id, :queue=>message.queue.url, :visibility_timeout=>message.queue.visibility_timeout}}
-  let(:pool) { double("pool") }
-  let(:sqs) { double('AWS::SQS') }
-  let(:backoff_func) { nil }
-
-  before do
-    allow(AWS::SQS).to receive(:new).and_return(sqs)
-    allow(sqs).to receive(:queues) { queues }
-
-    allow(queues).to receive(:url_for) { queue_url }
-    allow(queues).to receive(:[]) { queue }
-    allow(queue).to receive(:receive_message) { message }
-    allow(pool).to receive(:empty!) { nil }
-  end
-
-  describe "consuming messages" do
-    let!(:consumer_run_for_one_message) { allow(consumer).to receive(:running?).and_return(true, false) }
-    let!(:messages_be_unique) { allow_any_instance_of(Chore::DuplicateDetector).to receive(:found_duplicate?).and_return(false) }
-    let!(:queue_contain_messages) { allow(queue).to receive(:receive_messages).and_return(message) }
-
-    it 'should configure sqs' do
-      allow(Chore.config).to receive(:aws_access_key).and_return('key')
-      allow(Chore.config).to receive(:aws_secret_key).and_return('secret')
-
-      expect(AWS::SQS).to receive(:new).with(
-        :access_key_id => 'key',
-        :secret_access_key => 'secret'
-      ).and_return(sqs)
-      consumer.consume
-    end
-
-    it 'should not configure sqs multiple times' do
-      allow(consumer).to receive(:running?).and_return(true, true, false)
-
-      expect(AWS::SQS).to receive(:new).once.and_return(sqs)
-      consumer.consume
+require_relative '../spec_helper'
+
+module Chore
```

Comment:
The goal was to share AWS API object doubles in an effort to reduce the cognitive overhead of reasoning about all the SQS-related specs together. I just cargo-culted `module Chore` because that's the context the specs were all running in. I will finish reading through your interrelated  comments and either make updates or ask more questions :)

Reply:
I'm not sure if I follow here.  What's the benefit in putting specs in a module?  As far as I'm aware, we don't do this in any of our codebases and I'm not sure I've ever seen that as a practice in rspec docs.  The `describe` blocks themselves create anonymous modules, so there isn't typically value in surrounding them with another module.

Reply:
The same-functionality-but-different-implementation fake objects between the publisher and consumer was confusing to me, and it seemed to make sense to unify the objects to reduce that confusion as well as prevent drift. As far as the enclosing module declaration in the specs themselves, that was pre-existing. Did you want me to remove them?

Reply:
Update: it was pre-existing in the _publisher_, and I guess I updated the other spec to match.